src/Uno.UI.Runtime.Skia.Wpf/Input/WpfKeyboardInputSource.cs

```diff
@@ -70,26 +72,32 @@ private void HostOnKeyUp(object sender, System.Windows.Input.KeyEventArgs args)
 				this.Log().Trace($"OnKeyPressEvent: {args.Key} -> {virtualKey}");
 			}
 
+			var scanCode = GetScanCode(args);
+
 			KeyUp?.Invoke(this, new(
 				"keyboard",
 				virtualKey,
 				GetKeyModifiers(args.KeyboardDevice.Modifiers),
 				new CorePhysicalKeyStatus
 				{
-					ScanCode = (uint)args.SystemKey,
+					ScanCode = scanCode,
 					RepeatCount = 1,
 				},
-				KeyCodeToUnicode((uint)args.SystemKey)));
+				KeyCodeToUnicode(scanCode)));
 		}
 		catch (Exception e)
 		{
 			Windows.UI.Xaml.Application.Current.RaiseRecoverableUnhandledException(e);
 		}
 	}
 
+	// WPF doesn't expose the scancode, but it exists internally
+	private static uint GetScanCode(KeyEventArgs args)
+		=> (uint)((int)(typeof(KeyEventArgs).GetProperty("ScanCode", BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Instance)?.GetValue(args) ?? 0));
```

Comment:
Let's make it clear when the property cant 'e found. It will othewise fail with a null ref without much details.

Reply:
@jeromelaban The normal keyboard event can always find this property. The PresentationCore(WPF) will create the KeyEventArgs with the ScanCode, and the WPF will use the lParam windows message or MapVirtualKey to get the ScanCode.

And the wpf will only use the ScanCode property in `KeyDown --> Alt Numpad` business. In other words, the ScanCode property work in a less important piece of business logic code.

When the property can not be found? 1. The keyboard event from windows message, but wpf call the MapVirtualKey with Win32Exception. 2. We fake the keyboard event.

Reply:
The code to get the ScanCode (only ways) : https://github.com/dotnet/wpf/blob/187be47c5554d4745e9b17385889a8086d6e4286/src/Microsoft.DotNet.Wpf/src/PresentationCore/System/Windows/InterOp/HwndSource.cs#L2362

The code to use ScanCode (only ways) : https://github.com/dotnet/wpf/blob/187be47c5554d4745e9b17385889a8086d6e4286/src/Microsoft.DotNet.Wpf/src/PresentationCore/System/Windows/Input/TextCompositionManager.cs#L435

Reply:
The point here that [ScanCode](https://github.com/dotnet/wpf/blob/187be47c5554d4745e9b17385889a8086d6e4286/src/Microsoft.DotNet.Wpf/src/PresentationCore/System/Windows/Input/KeyEventArgs.cs#L206) is an internal property that may be changing names in the future. It's a best practice, when depending on internal properties, to validate that the property still exists and if not, to provide a proper error message.

Reply:
@jeromelaban Is the ScanCode here a necessary and important attribute in the UNO framework? If so, I'd be more than happy to submit an API change request in the WPF repository.

Reply:
According to @ramezgerges, it is important. Regardless, since we support .NET 7, we'll need to use reflection for the time being.

Reply:
> @jeromelaban Is the ScanCode here a necessary and important attribute in the UNO framework? If so, I'd be more than happy to submit an API change request in the WPF repository.

@jeromelaban I added here: https://github.com/dotnet/wpf/issues/8473